pkg/validator/shoot_validator.go

```diff
@@ -87,6 +87,19 @@ func (v *Shoot) validateShoot(valContext *validationContext) field.ErrorList {
 	allErrors = append(allErrors, gcpvalidation.ValidateWorkers(valContext.shoot.Spec.Provider.Workers, workersPath)...)
 	allErrors = append(allErrors, gcpvalidation.ValidateControlPlaneConfig(valContext.controlPlaneConfig, allowedZones, workersZones(valContext.shoot.Spec.Provider.Workers), controlPlaneConfigPath)...)
 
+	// WokerConfig
+	for i, worker := range valContext.shoot.Spec.Provider.Workers {
+		workerFldPath := workersPath.Index(i)
+		for _, volume := range worker.DataVolumes {
+			workerConfig, err := decodeWorkerConfig(v.decoder, worker.ProviderConfig)
```

Comment:
See https://github.com/gardener/gardener-extension-provider-gcp/pull/71#discussion_r415571266 - I think at this place it makes sense to check that the provider config may only be present if only `SCRATCH` volumes are used as data volumes.

Reply:
Actually, if there is not `WorkerConfig`, the `decodeWorkerConfig` method returns `nil`. In `ValidateWorkerConfig` method will check the logic.

Reply:
Hm, probably I didn't make it clear enough. I suggest to prevent such misconfigurations:

```yaml
workers:
- name: foo
  ...
  dataVolumes:
  - type: pd-ssd
  - type: SCRATCH
  providerConfig:
    apiVersion: aws.provider.extensions.gardener.cloud/v1alpha1
    kind: WorkerConfig
    volume:
      interface: foo
```

Basically, providing the worker config does only make sense if ONLY data volumes of `SCRATCH` type are provided. I guess this should be validated somewhere here in this file/during the loop over the data volumes.

Reply:
If it is only `pd-ssd`, i think it is better to ignore the `interface` to replace an exception.

I will validate the `interface` if the type of data volume is `SCRATCH `
```
if volumeType != nil && *volumeType == "SCRATCH" {
		if workerConfig == nil || workerConfig.Volume == nil || workerConfig.Volume.LocalSSDInterface == nil {
			allErrs = append(allErrs, field.Required(field.NewPath("volume", "localSSDInterface"), "must be set when using SCRATCH volumes"))
		} else {
			if !validVolumeLocalSSDInterfacesTypes.Has(*workerConfig.Volume.LocalSSDInterface) {
				allErrs = append(allErrs, field.NotSupported(field.NewPath("volume", "localSSDInterface"), *workerConfig.Volume.LocalSSDInterface, validVolumeLocalSSDInterfacesTypes.List()))
			}
		}
	}
```

When creating a `GCPMachineClass`,  the interface will be set when the type is `SCRATCH `, else the interface will be ignored.
```
if volume.Type != nil && *volume.Type == "SCRATCH" && workerConfig.Volume != nil && workerConfig.Volume.LocalSSDInterface != nil {
				disk["interface"] = *workerConfig.Volume.LocalSSDInterface
			}
```

Reply:
Agreed, especially as the `WorkerConfig` may evolve and contain way more fields in the future. 👍